agir/notifications/components/NotificationSettings/NotificationSettings.js

```diff
@@ -21,22 +25,54 @@ const NotificationSettings = (props) => {
     isValidating,
   } = useSWR(api.ENDPOINT.getSubscriptions);
 
+  const user = useSelector(getUser);
   const [isLoading, setIsLoading] = useState(false);
 
-  const notifications = useMemo(
-    () => getAllNotifications(groupData?.groups),
-    [groupData]
-  );
+  const { data: profile, mutate: mutateProfile } = useSWR("/api/user/profile/");
+
+  const notifications = useMemo(() => {
+    return getAllNotifications(groupData?.groups, user);
+  }, [groupData, user]);
+
+  const activeNewsletters = useMemo(() => {
+    return getNewsletterStatus(profile?.newsletters);
+  }, [profile]);
 
   const activeNotifications = useMemo(
     () => getNotificationStatus(userNotifications),
     [userNotifications]
   );
 
+  const allActiveNotifications = {
+    ...activeNewsletters,
+    ...activeNotifications,
+  };
+
   const handleChange = useCallback(
     async (notification) => {
       setIsLoading(true);
       let result;
+
+      // Update profile newsletters
+      if (notification.isNewsletter) {
+        if (notification.action === "add") {
+          result = await updateProfile({
+            newsletters: [...profile?.newsletters, notification.id],
+          });
+        }
+        if (notification.action === "remove") {
+          result = await updateProfile({
+            newsletters: profile?.newsletters.filter(
+              (notif) => notif !== notification.id
+            ),
+          });
+        }
+        setIsLoading(false);
+        mutateProfile();
+        return;
+      }
```

Comment:
Il n'y a pas le nouveau profil dans `result` ? Sinon, il n'y a pas forcément besoin de refaire une requête ici.
```suggestion
      if (notification.isNewsletter) {
        const newsletters = notification.action === "add" 
          ?  [...profile.newsletters, notification.id]
          :   profile.newsletters.filter(
                (notif) => notif !== notification.id
              );
        result = await updateProfile({ newsletters });
        setIsLoading(false);
        mutateProfile(userProfile => result?.data || { ...userProfile, newsletters }), false);
        return;
      }
```